pkg/repo/chartrepo.go

```diff
@@ -115,12 +115,16 @@ func (r *ChartRepository) Load() error {
 
 // DownloadIndexFile fetches the index from a repository.
 func (r *ChartRepository) DownloadIndexFile() (string, error) {
-	indexURL, err := ResolveReferenceURL(r.Config.URL, "index.yaml")
+	jsonIndexURL, err := ResolveReferenceURL(r.Config.URL, "index.json")
+	if err != nil {
+		return "", err
+	}
+	yamlIndexURL, err := ResolveReferenceURL(r.Config.URL, "index.yaml")
 	if err != nil {
 		return "", err
 	}
 
-	resp, err := r.Client.Get(indexURL,
+	resp, err := r.Client.Get(jsonIndexURL,
```

Comment:
If the file does not exist (HTTP 404), this would return an error and it would not fallback to loading the `index.yaml`. See also https://github.com/helm/helm/blob/4e447d87cd3222aa5e5ea2b38170e9d80b99c0b4/pkg/getter/httpgetter.go#L91-L93

In addition, on a successful retrieval any parsing error would be swallowed by an attempt to fetch and load the `index.yaml` (which may not exist). Yielding absolutely no information to the user to figure out what went wrong.

Given this, I think it would be better to not perform a fallback based on loading errors. But to rather fall back based on the error from `r.Client.Get`, which may require introducing a typed error in `httpgetter.go` to distinguish a HTTP 404 from transient errors.